src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -19,9 +26,42 @@ public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bind
             return base.BindModelAsync(bindingContext);
         }
 
+        /// <inheritdoc />
         protected override object GetModel(IEnumerable<TElement> newCollection)
         {
+            if (newCollection == null)
+            {
+                return null;
+            }
+
             return newCollection.ToArray();
         }
+
+        /// <inheritdoc />
+        protected override void CopyToModel([NotNull] object target, IEnumerable<TElement> sourceCollection)
+        {
+            TElement[] targetArray = target as TElement[];
+            Debug.Assert(targetArray != null); // This binder is instantiated only for array model types.
+
+            if (sourceCollection != null && targetArray != null)
```

Comment:
will be `null` when form data includes a prefix match but value provider returns `null`.  not a common case but can happen.